### PR TITLE
Refactor utils command to follow Complete/Validate/Run pattern

### LINKED ISSUE #1194: Refactor utils command to follow complete-validate-run pattern
more info in #1142

### PR DESCRIPTION
**Pull request by metacosm**
## Was the change discussed in an issue?
fixes #1194 

## How to test changes?
Play with `odo utils` commands.

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -395,3 +398,43 @@ func (c *ConfigInfo) DeleteProject(projectName string) error {
 	}
 	return nil
 }
+
+var (
+	// records information on supported parameters
+	supportedParameterDescriptions = map[string]string{
+		"UpdateNotification": "Controls if an update notification is shown or not (true or false)",
```
Comment (anmolbabu) [inline pkg/config/config.go]: Can this be configurables be string constants?
I see them being referred in more than 1 places which might be prone to bugs
So, this then converts to map[string_alias]string

File: pkg/config/config.go [hunk 2]
```diff
@@ -141,29 +142,31 @@ func (c *ConfigInfo) writeToFile() error {
 // SetConfiguration modifies Odo configurations in the config file
 // as of now being used for nameprefix, timeout, updatenotification
 func (c *ConfigInfo) SetConfiguration(parameter string, value string) error {
-	// processing values according to the parameter names
-	switch parameter {
+	if p, ok := asSupportedParameter(parameter); ok {
```
Comment (anmolbabu) [inline pkg/config/config.go]: I tend to feel config or configurable would be apt instead of parameter?

Comment (anmolbabu) [reply pkg/config/config.go]: but from what i see in git diff view, its been like that for some time now in which case please ignore my comment

Comment (metacosm) [reply pkg/config/config.go]: I actually have more ideas on how to refactor things here but I don't think it was appropriate to do these changes in this PR (I've deleted code that I spent a couple of hours on with some of these changes I had in mind and decided against pushing further).

File: pkg/config/config.go (hunk 1 above)
Comment (metacosm) [reply pkg/config/config.go]: Let me see what I can do.

Comment (metacosm) [reply pkg/config/config.go]: Extracted constants.

File: tests/e2e/e2e_test.go [hunk 3]
```diff
@@ -80,22 +80,24 @@ var _ = Describe("odoe2e", func() {
...
 			configOutput := runCmd("odo utils config view")
 			Expect(configOutput).To(ContainSubstring("true"))
-			Expect(configOutput).To(ContainSubstring("UpdateNotification"))
+			Expect(configOutput).To(ContainSubstring(config.UpdateNotificationSetting))
+			Expect(configOutput).To(ContainSubstring(config.NamePrefixSetting))
+			Expect(configOutput).To(ContainSubstring(config.TimeoutSetting))
 		})
 		It("should be checking to see if timeout is the same as the constant", func() {
-			configOutput := runCmd("odo utils config view|grep Timeout")
+			configOutput := runCmd("odo utils config view|grep " + config.TimeoutSetting)
```
Comment (amitkrout) [inline tests/e2e/e2e_test.go]: Avoid using terminal specific command like ```grep```. regular expression would be an alternate choice and can be run on any terminal.

Comment (metacosm) [reply tests/e2e/e2e_test.go]: I didn't create these tests, they've been like this for a while, not going to change them now.

Comment (amitkrout) [reply tests/e2e/e2e_test.go]: Ok, will track this change in #1129 then.

**Comment by metacosm** [comment]
> Major command `odo utils config set ...` is missed as part of this pr. Would you please add scenario for `odo utils config set ...` in integration test too.
> 
> ```
> $ ./odo utils -h
> Utilities for terminal commands and modifying Odo configurations
> 
> Usage:
>   odo utils [command]
> 
> […]
> 
>    # Set a configuration value
>    odo utils config set UpdateNotification false
>    odo utils config set NamePrefix "app"
>    odo utils config set Timeout 20
> 	
>   # For viewing the current configuration
>    odo utils config view
>   
> […]
> ```

Actually, `set` is covered: https://github.com/redhat-developer/odo/blob/master/tests/e2e/e2e_test.go#L92-L99

**Pull request merged**